onnxruntime/contrib_ops/cpu/transformers/beam_search_parameters.cc

```diff
@@ -123,8 +123,18 @@ void BeamSearchParameters::ParseFromInputs(OpKernelContext* context) {
   logits_processor = logits_processor_tensor ? static_cast<int>(*logits_processor_tensor->Data<int32_t>()) : 0;
   ORT_ENFORCE(logits_processor >= 0,
               "logits_processor shall be a non-negative integer, got ", logits_processor);
-}
 
+  auto* temperature_tensor = context->Input<Tensor>(14);
```

Comment:
Guard this section by `if (this->model_type == IGenerationParameters::kModelTypeWhisper){}` since normal BeamSearch op does not have that.